packages/cells/src/widget.ts

```diff
@@ -253,13 +269,37 @@ class Cell extends Widget {
   }
 
   /**
-   * The view state of input being collapsed.
+   * The view state of input being hidden.
    */
-  get inputCollapsed(): boolean {
-    return this._inputCollapsed;
+  get inputHidden(): boolean {
+    return this._inputHidden;
   }
-  set inputCollapsed(value: boolean) {
-    this._inputCollapsed = value;
+  set inputHidden(value: boolean) {
+    if (this._inputHidden === value) {
+      return;
+    }
+    let layout = this._inputWrapper.layout as PanelLayout;
+    if (value) {
+      layout.removeWidget(this._input);
```

Comment:
We should not be calling `removeWidget` directly, per the [docs](https://github.com/phosphorjs/phosphor/blob/06f24b7429e79da108e909f4aa1ba40661fe9372/packages/widgets/src/layout.ts#L175).  Can this be handled with show/hide?